### PR TITLE
Integration fixes

### PR DESCRIPTION
**Pull request by jgraff2**
These are fixes for the bugs that were revealed by the integration tests, plus some structural changes for testability.

Since this is a collection of unrelated changes, I have left comments explaining why each was necessary.

### CODE REVIEW

File: src/heartbeat/utils.js [hunk 1]
```diff
@@ -62,7 +58,20 @@ function changeCollectorStatus(currentStatus, newStatus) {
...
+    if (oldValue && newValue !== oldValue) {
+      if (key === 'heartbeatIntervalMillis') {
+        repeater.updateHeartbeatRepeater(newValue);
+      } else if (key === 'sampleUpsertQueueTimeMillis') {
+        queue.updateFlushTimeoutAll(newValue);
+      } else if (key === 'maxSamplesPerBulkUpsert') {
+        queue.updateSizeAll(newValue);
+      }
+    }
+  });
```
Comment (jgraff2) [inline src/heartbeat/utils.js]: Need to update existing repeaters/queues when the config is updated in the heartbeat.

File: src/heartbeat/utils.js [hunk 2]
```diff
@@ -62,7 +58,20 @@ function changeCollectorStatus(currentStatus, newStatus) {
  */
 function updateCollectorConfig(cc) {
   const config = configModule.getConfig();
-  Object.keys(cc).forEach((key) => config.refocus[key] = cc[key]);
+  Object.keys(cc).forEach((key) => {
+    const oldValue = config.refocus[key];
+    const newValue = cc[key];
+    config.refocus[key] = cc[key];
+    if (oldValue && newValue !== oldValue) {
```
Comment (cassiodias) [inline src/heartbeat/utils.js]: Any special reason of oldValue && newValue... isn't be the same as newValue != oldValue?

File: src/remoteCollection/handleCollectResponse.js [hunk 3]
```diff
@@ -100,92 +101,79 @@ function prepareTransformArgs(generator) {
...
-    for (let subject of g.subjects) {
-      const _g = JSON.parse(JSON.stringify(g));
-      _g.subjects = [subject];
-      let qLength = handleCollectResponse(prepareRemoteRequest(_g));
-
-      // return qLength if this is the final subject we enqueue
-      if (subject === g.subjects[g.subjects.length - 1]) return qLength;
-    }
-  });
+    const _g = JSON.parse(JSON.stringify(generator));
```
Comment (cassiodias) [inline src/remoteCollection/handleCollectResponse.js]: Is there any code convention when using underscore for naming vars?

Comment (jgraff2) [reply src/remoteCollection/handleCollectResponse.js]: It's sometimes used to indicate a temporary variable when the variable name is already taken.
This line was already there, I just changed the enclosing loop.

**Pull request merged**